presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcPageSink.java

```diff
@@ -165,6 +165,11 @@ else if (javaType == Slice.class) {
             throw new PrestoException(JDBC_NON_TRANSIENT_ERROR, e);
         }
         catch (SQLException e) {
+            SQLException nextExc = e.getNextException();
+            while (nextExc != null) {
+                e.addSuppressed(nextExc);
```

Comment:
be defensive here, since this code is (and will not be) well test covered:

```suggestion
                if (e != nextExc) {
                    e.addSuppressed(nextExc);
                }
```

Reply:
Also, it might be good to distinguish these:

```
e.addSuppressed(new Exception("Next exception", nextException));

Reply:
A better defense would be breaking out of the loop when `e == nextException`. If you see the original exception again, you're in a loop. You could end up in a loop without the original, too, which would require a few more steps to check.

Reply:
i didn't think about this. Do you have evidence loop is possible? Or just to be on the safe side?

Reply:
It's technically _possible_ (`x.setNextException(x)`), but it really shouldn't ever happen in a sane JDBC driver, and it doesn't seem like the sort of thing that would easy to do by accident.

(But if we're going to have the check for the original exception, we should break rather than skip it.)